### DISCUSSION TITLE
create-wmr with typescript support or example

### DISCUSSION THREAD

**Discussion by wilberforce** [preactjs/wmr#891] [Q&A]
Is there boiler plate example that shows a simple app that uses ts rather then js?

Thanks

**Comment by rschristian** [comment]
The template ships with a `tsconfig.json`, so you'll get full TS type checking as-is. If you want to author in TS, all you have to do is change the file extensions. `*.js` -> `*.tsx`

I believe the only other alteration you could want to make is the `About` component:

https://github.com/preactjs/wmr/blob/3401a9bfa6491d25108ad68688c067a7e17d0de5/packages/create-wmr/tpl/public/pages/about/index.js#L3

You could type this as the following:

```tsx
 const About = ({ query }: { query: Record<string, string> }) => ( 
```

**Reply by wilberforce** [reply]
Thanks - so wmr does transpile on the fly?

**Reply by rschristian** [reply]
"on the fly" -- depends on how you're defining that. There's nothing "on the fly" about using the output of `wmr build`. All AOT at that point.

But yes, it transpiles TS using [sucrase](https://github.com/alangpierce/sucrase).

**Reply by wilberforce** [reply]
It's now decorators that are tripping me up.

```
./public\motion-todos.ts - Unexpected character '@' (./motion-todos.ts:26:1)

  24 |
  25 |
> 26 | @customElement('motion-todos')
     | ^
```

I have added `"experimentalDecorators": true,` to tsconfig.js

However, is there something that needs to be added to wmr.config?

**Reply by rschristian** [reply]
I don't believe decorators are supported at the moment. Want to make an issue so that this can be better tracked?

Your `tsconfig.json` is pretty much just for your editor. I believe WMR will only use `"paths"`. If you have it set that is.

**Reply by wilberforce** [reply]
thanks - I will do that
